src/parameters.jl

```diff
@@ -149,3 +149,43 @@ function flatten(::Type{T}, x::Deferred) where T<:Real
     unflatten_Deferred(v_new::Vector{T}) = Deferred(x.f, unflatten(v_new))
     return v, unflatten_Deferred
 end
+
+"""
+    nearest_orthogonal_matrix(X::StridedMatrix)
+
+Project `X` onto the closest orthogonal matrix in Frobenius norm.
+
+Originally used in varz: https://github.com/wesselb/varz/blob/master/varz/vars.py#L446
+"""
+@inline function nearest_orthogonal_matrix(X::StridedMatrix)
+    # Inlining necessary for type inference for some reason.
+    U, _, V = svd(X)
+    return U * V'
+end
+
+"""
+    orthogonal(X::StridedMatrix{<:Real})
+
+Produce a parameter whose `value` is constrained to be positive. The argument `X` need not
+be orthogonal.
+
+This functionality projects `X` onto the nearest element subspace of orthogonal matrices (in
+Frobenius norm) and is overparametrised as a consequence.
```

Comment:
What do you mean here by 'overparameterised as a consequence'?

Reply:
If `size(X) = (N, M)`, then it has `NM` free parameters. Conversely, its orthogonal projection has only `N(M-1)` parameters because each row is normalised.